jwst/associations/lib/rules_level2b.py

```diff
@@ -507,9 +507,29 @@ def __init__(self, *args, **kwargs):
                 sources=['filter'],
                 value='opaque'
             ),
-            DMSAttrConstraint(
-                name='opmode',
-                sources=['opmode'],
+            Constraint(
+                [
+                    Constraint(
+                        [DMSAttrConstraint(
+                            name='opmode',
```

Comment:
So to put this logic in simple terms, this rule passes if either: the OPMODE is not MSASPEC - or - OPMODE is MSASPEC and MSAMETFL is present?